GraphExplorerPermissionsService/Services/PermissionsStore.cs

```diff
@@ -520,11 +475,45 @@ private static string CleanRequestUrl(string requestUrl)
                         .ToLowerInvariant();
         }
 
+        /// <summary>
+        /// Retrieves the scopes information for a given list of scopes.
+        /// </summary>
+        /// <param name="scopesInformationDictionary">The source of the scopes information.</param>
+        /// <param name="scopes">The target list of scopes.</param>
+        /// <param name="scopeType">The type of scope from which to retrieve the scopes information for.</param>
+        /// <returns>A list of <see cref="ScopeInformation"/>.</returns>
+        private static List<ScopeInformation> GetScopesInformation(IDictionary<string, IDictionary<string, ScopeInformation>> scopesInformationDictionary,
```

Comment:
if the inputs will not be modified, make them as read only such as `IReadOnlyDictionary` and `IReadOnlyList`

Reply:
These changes are from an upstream merge and not mine, so I cant change it.